sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/datasources/subscriptions/SubscriptionsDbConnector.groovy

```diff
@@ -1,25 +1,118 @@
 package life.qbic.portal.sampletracking.datasources.subscriptions
 
+import groovy.util.logging.Log4j2
+
 import life.qbic.business.project.subscribe.Subscriber
 import life.qbic.business.project.subscribe.SubscriptionDataSource
 
+import life.qbic.business.DataSourceException
+import life.qbic.portal.sampletracking.datasources.database.ConnectionProvider
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+
 /**
- * <b><short description></b>
+ * <b>Provides methods to handle user's subscriptions to projects</b>
  *
- * <p><detailed description></p>
+ * <p>Connects to the subscription database in order to store or fetch user's subsriptions to project. Uses Subscriber objects and the project code</p>
  *
- * @since <version tag>
+ * @since 1.0.0
  */
 class SubscriptionsDbConnector implements SubscriptionDataSource {
+  
+  private final ConnectionProvider connectionProvider
+  
+      /**
+       * Creates a database connector that will connect to the database using the provided connection
+       * provider.
+       * @param connectionProvider the connection provider providing the connections that are used to
+       * connect to the database
+       * @since 1.0.0
+       */
+      SubscriptionsDbConnector(ConnectionProvider connectionProvider) {
+          this.connectionProvider = connectionProvider
+      }
+
     /**
-     * Creates the subscriber in the data source and subscribes him to the project.
+     * Creates the subscriber in the data source, if they don't exist, and subscribes them to the project.
      * @param subscriber
      * @param projectCode
      * @since 1.1.0
      */
     @Override
     void subscribeToProject(Subscriber subscriber, String projectCode) {
-        //TODO implement
-        println "subscriber = $subscriber, projectCode = $projectCode"
+          try {
+            Connection connection = connectionProvider.connect()
+            connection.setAutoCommit(false)
+      
+            connection.withCloseable { it ->
+              try {
+                int subscriberId = getSubscriberId(it, subscriber)
+                addSubscription(it, subscriberId, projectCode)
+                connection.commit()
+              } catch (Exception e) {
+                log.error(e.message)
+                log.error(e.stackTrace.join("\n"))
```

Comment:
to log you need to add @Log4j2 to the top of the class or in general define the log variable ;)